src/backend/pipeline/modules/mosaicing.py

```diff
@@ -30,10 +29,12 @@ def run(self, data: Data) -> Data:
         """
 
         self.prepare(data)
-        
         # Check if there are multiple input images
         if len(data.input) == 1:
             data.modules[self.type]["stitched"] = data.input[0]
+            patches = self.create_patches(data.input[0], data.input[0].channels) 
+            data.modules[self.type]["patches"] = patches
+            data.persistable[self.type]["stitched"] = data.input[0]
```

Comment:
This is problematic if you mutate the output. I think a better approach is to mark persistable entities rather than duplicate them. You can do that in `prepare` when you're preparing the data object for population and then save all persistable entities after you've run the module.